File: Common Divisors/common_divisors.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

vector<string> check_repeated(string s)
{
    ll len = s.length();
    vector<int> factors;
    vector<string> repeated;
    for(ll i = 1 ;i <= len;i++ )
    {
        if(len % i == 0)
        factors.push_back(i);
    }
    
    for (auto i = factors.begin(); i != factors.end(); ++i)
    {   string s1 = "";
        for(int j = 0 ;j < *i ;j++ )
         s1 += s[j];
        bool found = true; 
        ll j = *i ;
        while (j < len)
        {   string s2 = "";
            for(int z = j;z < j + *i;z++)
              s2 += s[z];
              
            if(s1 != s2)
            {
                found = false;
                break;
            }
              
            j += *i ;
        }
        
        if(found)
        repeated.push_back(s1);
         
    }
    return (repeated);
    
    
    
}

int Common_Divisors()
{ string first,second;
  cin>>first>>second;
  vector<string> repeated_first = check_repeated(first);
  vector<string> repeated_second = check_repeated(second);
  ll common_divisors = 0;
  
  
  for(auto i = repeated_first.begin();i != repeated_first.end();++i)
  {for(auto j = repeated_second.begin();j != repeated_second.end();++j)
       {if(*i == *j)
           { common_divisors++;
               break;
           }
       }
  }
  return (common_divisors);
  
  
  
  
}


int main() {
	int remained_test_cases = 1 ;
// 	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   cout<<Common_Divisors();
	    remained_test_cases--;
	}
	return 0;
}

File: Common Divisors/common_divisors.py
def check_repeatation(string):
    l1 = len(string)
    factors = []
    for i in range(1,l1+1):
        if l1 % i == 0:
            factors.append(i)
    repeated = []        
    for i in range(len(factors)):
        found = True
        s1 = string[:factors[i]]
        j = factors[i]
        while j < l1:
            s2 = string[j:j+factors[i]]
            j += factors[i]
            if s1 != s2:
                found = False
                break 
        if found:
            repeated.append(s1)
            
    return repeated        
            
        
        


def Common_Divisors():
    stirng1 = input()
    string2 = input()
    s1_repeated = check_repeatation(stirng1)
    s2_repeated = check_repeatation(string2)
    
    common_divisor = 0 
    for i in range(len(s1_repeated)):
        if s1_repeated[i] in s2_repeated:
            common_divisor += 1 
    return common_divisor        
            
        
remained_test_cases = 1 
# remained_test_cases = int(input())
while remained_test_cases > 0:
    print(Common_Divisors())
    remained_test_cases -= 1 